Add getting-started example for create-react-app (#920)

RobJacobson opened the pull request:
Here's a first attempt at adding a minimal example using create-react-app.  How does this look?

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/react-map-gl?pullRequest=1003) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/react-map-gl?pullRequest=1003) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/react-map-gl?pullRequest=1003) it.</sub>

RobJacobson commented:
Hmm... I'm confused about the Travis error.  When I run the tests in my CLI, I get this error message (which isn't terribly helpful):

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! react-map-gl@5.2.0 test: `flow check && ocular-test`
npm ERR! Exit status 1

Have I made a mistake or omitted something necessary?  (This is my first pull request here.)  Thanks in advance.

Pessimistress commented on examples/get-started/create-react-app/package.json:
```diff
@@ -0,0 +1,35 @@
+    "eject": "react-scripts eject"
+  },
+  "eslintConfig": {
+    "extends": "react-app"
```
This is breaking CI. You need to either remove this or add `eslint-config-react-app` to the root `devDependencies`.

RobJacobson replied:
Thanks. I added the dev dependency.

RobJacobson replied:
Thanks. I added eslint-config-react-app.

Pessimistress replied:
You need to add it to the root package.json. CI does not install dependencies for each example.

Pessimistress commented on examples/get-started/create-react-app/src/map.jsx:
```diff
@@ -0,0 +1,47 @@
+        width="100%"
+        height="100vh"
+        mapboxApiAccessToken={API_KEY}
+        onViewportChange={viewport => this.setState({ viewport })}
```
`this.updateViewport`

RobJacobson replied:
Good catch!

Pessimistress commented on examples/get-started/create-react-app/src/map.jsx:
```diff
@@ -0,0 +1,47 @@
+      bearing: 0,
+      pitch: 0
+    },
+    popupInfo: null
```
This doesn't seem to be used

RobJacobson replied:
True.  (A copy-paste error.)

RobJacobson replied:
Good eye!  Sorry for the copy-paste error.

RobJacobson commented:
I've made the requested changes.  However, I'm still failing the Travis CI check.  Can you provide any guidance?

Pessimistress commented on examples/get-started/create-react-app/.gitignore:
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Do you need this file specifically for this example? Can this be rolled into the root .gitignore?

Pessimistress commented:
> I've made the requested changes. However, I'm still failing the Travis CI check. 

You need to add it to the root package.json. CI does not install dependencies for each example.

RobJacobson commented:
I'm terribly sorry, but the Travis build failed again.  I removed "eslint-config-react-app" from the example's config.json file and moved it to config.json in the root project.  Could you please provide a further hint?  (The Travis CI check seems like a "black box" to me.)

Pessimistress commented:
> I removed "eslint-config-react-app" from the example's config.json file and moved it to config.json in the root project.

CI runs `yarn` at root. If `eslint-config-react-app` is not a root dependency, then it won't be installed.

If you don't really need the eslint setup for your example, then just remove it period. The repo has its own eslint configuration.

Pessimistress commented:
Looks like you have committed some irrelevant files in `examples/geojson-markers`.

RobJacobson commented:
Thank you.  I've done the following:
1. Deleted esLintConfig from package.json for the CRA example.
2. Deleted eslint-config-react-app from root/package.json.
3. Deleted the extraneous files.  (Sorry about that.)

Unfortunately, the Travis check is still failing.  I'm feeling like an idiot.  Could you please elaborate again on the problem?

Pessimistress commented:
CI is failing for a different issue (lint error). 

```
/home/travis/build/uber/react-map-gl/examples/get-started/create-react-app/src/index.js
  5:26  error  'document' is not defined  no-undef
```

Have you run `yarn` at root before you commit?

RobJacobson commented:
Thanks for your tip.  I'm using VS Code with the ESLint extension.  I'm getting the following error message at the top level of the project:

```
ESLint: ESLint configuration in .eslintrc.js is invalid: - Property "overrides" is the wrong type (expected array but got `{"files":["*.spec.js","webpack.config.js","**/bundle/*.js"],"rules":{"import/no-extraneous-dependencies":0}}`). . Please see the 'ESLint' output channel for details.
```

What am I doing wrong?  Thanks for your patience with this issue.

Pessimistress commented:
This issue is recently fixed, you may need to rebase master.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/uber/react-map-gl?pullRequest=1003) <br/>All committers have signed the CLA.
